### PR TITLE
Removed auto type check on buffer write and enter

### PR DESCRIPTION
**Pull request by 2chilled**
Addresses https://github.com/ensime/ensime-vim/issues/209 but removes the auto type check also on buffer enter

### CODE REVIEW

File: ensime_shared/ensime.py [hunk 1]
```diff
@@ -936,8 +936,6 @@ def buffer_leave(self, filename):
 
     def buffer_enter(self, filename):
```
Comment (ChrisCoffey) [inline ensime_shared/ensime.py]: Please remove this hook as well.

**Comment by ChrisCoffey** [comment]
Please update the documentation with the VimL auto command to provide this behavior for users that want it.

**Comment by ChrisCoffey** [comment]
Code looks good, just needs the documentation.

**Comment by ChrisCoffey** [comment]
LGTM. @jvican merge when you're ready

**Comment by jvican** [comment]
LGTM, thanks for the contribution!

**Pull request merged**